File: libmvp/src/main/java/com/mr/k/mvp/base/IBaseMvpPresenter.java
package com.mr.k.mvp.base;

import com.trello.rxlifecycle2.LifecycleProvider;

/*
 * created by Cherry on 2019-12-20
 **/
public interface IBaseMvpPresenter<V extends IBaseMvpView> {

    void attachView(V view);

    void detachView();

    LifecycleProvider getProvider();

}

File: libmvpkotlin/src/main/java/com/mr/k/mvp/base/IBaseMvpPresenter.kt
package com.mr.k.mvp.base

import com.trello.rxlifecycle2.LifecycleProvider

/*
 * created by Cherry on 2019-12-20
 **/
interface IBaseMvpPresenter<V : IBaseMvpView<*>> {

    fun attachView(view: V)
    fun detachView()



}